src/reports_update/reports_updater.rs

```diff
@@ -53,24 +54,44 @@ pub trait SignedReportExt {
 }
 impl SignedReportExt for SignedReport {}
 
-pub struct ReportsUpdater<'a, T: Preferences, U: TcnDao, V: TcnMatcher, W: TcnApi, X: MemoMapper> {
+pub struct ReportsUpdater<
+    'a,
+    T: Preferences,
+    U: TcnDao,
+    V: TcnMatcher,
+    W: TcnApi,
+    X: MemoMapper,
+    Y: AlertDao,
+> {
     pub preferences: Arc<T>,
     pub tcn_dao: Arc<U>,
     pub tcn_matcher: V,
     pub api: &'a W,
     pub memo_mapper: &'a X,
     pub exposure_grouper: ExposureGrouper,
+    pub alert_dao: Arc<Y>,
 }
 
-impl<'a, T, U, V, W, X> ReportsUpdater<'a, T, U, V, W, X>
+impl<'a, T, U, V, W, X, Y> ReportsUpdater<'a, T, U, V, W, X, Y>
 where
     T: Preferences,
     U: TcnDao,
     V: TcnMatcher,
     W: TcnApi,
     X: MemoMapper,
+    Y: AlertDao,
 {
-    pub fn fetch_new_reports(&self) -> Result<Vec<Alert>, ServicesError> {
+    pub fn update_and_fetch_alerts(&self) -> Result<Vec<Alert>, ServicesError> {
```

Comment:
Alternative names: 

- fetch_updated_alerts  
- get_fresh_alerts

Reply:
`update_and_fetch_alerts` may not sound elegant, but seems clearer about what it does.